### PR TITLE
Added contributing details

### PR DESCRIPTION
**Pull request by Andrew-Chen-Wang**
Adds details for how to contribute. Also, this is meant to help #2.

* Added requirements file for developers
* Added notes on how to setup workstation in README.rst
* Added several kinds of environments for developers in .gitignore

TODO/FIXME:

 - [ ] Should we change the Travis build to use a matrix?

### CODE REVIEW

File: requirements-dev.txt [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# Core
+Cython>=3.0a4
+typing-extensions
```
Comment (dvarrazzo) [inline requirements-dev.txt:3]: There requirements get installed by `python setup.py develop`. Unfortunately I don't see how to specify the test ones (`python setup.py test` is deprecate). The dependencies are specified in `tox.ini`: I'll write something in the comments: please tell me if it makes sense.

**Comment by dvarrazzo** [comment]
In order to run all the tests, `tox -s` will do, but it's pretty slow, because it rebuilds the sdist every time. I have found the [tox-current-env](https://pypi.org/project/tox-current-env/) package which seems useful. Something like this works:

```
# create and activate a virtualenv: here, there, at leisure
virtualenv -p python3 env
source env/bin/activate

# install the dependencies
python setup.py develop

# install tox and all the testing dependencies
pip install tox tox-current-env
tox -s --print-deps-to-file requirements-dev.txt
pip install -r requirements-dev.txt

# now you can use tox to run all the tests
tox --current-env -s

# or just some
tox --current-env -e py37 -e black -e flake8 -e mypy
```

I wouldn't like to duplicate the dependencies in `tox.ini` into a `requirement-dev.txt` file, but this procedure can maybe be streamlined. Does it work for you? How would you document it?

File: .gitignore [hunk 2]
```diff
@@ -1,7 +1,18 @@
-env
```
[the remark this answers is not in the document]
Comment (dvarrazzo) [reply .gitignore:1]: done, I pushed it too with some other changes in this branch

**Comment by Andrew-Chen-Wang** [comment]
I like the tox-current-env package. That's pretty neat. I can understand the duplicate dependency thing, but it'd be much easier to simply pip install a requirements file and move on rather than having a few more lines to copy... however:

Trying to build from setup.py, install, or do that develop command, all results in a missing header file: endian.h. Perhaps that's resulting in my setup's error with datetime? My traceback:

```
running develop
running egg_info
writing psycopg3.egg-info/PKG-INFO
writing dependency_links to psycopg3.egg-info/dependency_links.txt
writing requirements to psycopg3.egg-info/requires.txt
writing top-level names to psycopg3.egg-info/top_level.txt
reading manifest file 'psycopg3.egg-info/SOURCES.txt'
reading manifest template 'MANIFEST.in'
writing manifest file 'psycopg3.egg-info/SOURCES.txt'
running build_ext
building 'psycopg3._psycopg3' extension
gcc -Wno-unused-result -Wsign-compare -Wunreachable-code -fno-common -dynamic -DNDEBUG -g -fwrapv -O3 -Wall -arch x86_64 -g -I/usr/local/include -I/Library/Frameworks/Python.framework/Versions/3.7/include/python3.7m -c psycopg3/_psycopg3.c -o build/temp.macosx-10.9-x86_64-3.7/psycopg3/_psycopg3.o
psycopg3/_psycopg3.c:794:10: fatal error: 'endian.h' file not found
#include <endian.h>
         ^~~~~~~~~~
1 error generated.
error: command 'gcc' failed with exit status 1
```

Perhaps it's an OS thing? I can try on a Linux machine sometime later (I'm currently using OS X, but perhaps it's just `psycopg3/types/endian.pxd` not working?). However, running pytest still works :P So...

**Comment by dvarrazzo** [comment]
`endian.h` is probably a Linux thing. You can [try to use `machine/endian.h`](https://stackoverflow.com/questions/20813028/endian-h-not-found-on-mac-osx) apparently. If this works send back a patch (with the `#ifdef` required to work on both systems). Without that it won't compile the C extension but the python code [works the same](https://www.varrazzo.com/blog/2020/05/19/a-trip-into-optimisation/).

**Comment by dvarrazzo** [comment]
I've added hacking info to the readme and tweaked tools config a bit. This sholud do but let me know if there is important info missing.

**Comment by Andrew-Chen-Wang** [comment]
@dvarrazzo Using the machine/endian.h import in endian.pxd worked... sorta. Build worked. Although it broke my setup (that I also have no idea how I set up in the first place). Was working fine... hmm

**Pull request closed without merging**